app/components/channel_icon/dm_avatar/index.ts

```diff
@@ -1,39 +1,29 @@
 // Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
 // See LICENSE.txt for license information.
 
-import {Q} from '@nozbe/watermelondb';
 import {withDatabase} from '@nozbe/watermelondb/DatabaseProvider';
 import withObservables from '@nozbe/with-observables';
 import {of as of$} from 'rxjs';
 import {switchMap} from 'rxjs/operators';
 
-import {MM_TABLES, SYSTEM_IDENTIFIERS} from '@constants/database';
+import {observeCurrentUserId} from '@queries/servers/system';
+import {observeUser} from '@queries/servers/user';
 import {getUserIdFromChannelName} from '@utils/user';
 
 import DmAvatar from './dm_avatar';
 
 import type {WithDatabaseArgs} from '@typings/database/database';
-import type SystemModel from '@typings/database/models/servers/system';
-import type UserModel from '@typings/database/models/servers/user';
-
-const {SERVER: {USER, SYSTEM}} = MM_TABLES;
-const {CURRENT_USER_ID} = SYSTEM_IDENTIFIERS;
 
 const enhance = withObservables(['channelName'], ({channelName, database}: {channelName: string} & WithDatabaseArgs) => {
-    const currentUserId = database.get<SystemModel>(SYSTEM).findAndObserve(CURRENT_USER_ID).pipe(
-        switchMap(({value}) => of$(value)),
-    );
+    const currentUserId = observeCurrentUserId(database);
 
     const authorId = currentUserId.pipe(
         switchMap((userId) => of$(getUserIdFromChannelName(userId, channelName))),
     );
 
     const author = authorId.pipe(
         switchMap((id) => {
-            return database.get<UserModel>(USER).query(Q.where('id', id)).observe().pipe(
-                // eslint-disable-next-line max-nested-callbacks
-                switchMap((users) => (users[0] ? of$(users[0]) : of$(undefined))),
-            );
+            return observeUser(database, id);
```

Comment:
based on what the observer function returns, `author` in this context can be `undefined` what I'm failing to see is the changes in the component props where this props can be undefined, is it cause TS is not complaining?

I wonder if we have:
1. make it so that `withObservables` complaints
2. go through all of this changes and see what are we passing to the actual component and mark them as optional appropriately and have the component be aware of it

Reply:
Regarding 1, I created a ticket one month ago here: https://github.com/Nozbe/withObservables/issues/123 No replies to it.
I can go through all the components making the change, but I wonder if we should defer that to a second PR. If we are able to "fix the types of withObservables", will make our lives easier, since it will yell at all the right places.